tests/QssTestDefs.js

```diff
@@ -22,37 +22,68 @@ var fluid = require("infusion"),
-    clickAppTextZoomBtn = "jQuery(\".flc-quickSetStrip > div:nth-child(3)\").click()",
-    clickReadAloudBtn = "jQuery(\".flc-quickSetStrip > div:nth-child(5)\").click()",
-    clickMoreBtn = "jQuery(\".flc-quickSetStrip > div:nth-last-child(6)\").click()",
-    clickSaveBtn = "jQuery(\".flc-quickSetStrip > div:nth-last-child(5)\").click()",
-    clickUndoBtn = "jQuery(\".flc-quickSetStrip > div:nth-last-child(4)\").click()",
-    clickResetAllBtn = "jQuery(\".flc-quickSetStrip > div:nth-last-child(3)\").click()",
-    clickPspBtn = "jQuery(\".flc-quickSetStrip > div:nth-last-child(2)\").click()",
+var hoverCloseBtn = "jQuery(\".flc-quickSetStrip > div:last-of-type\").trigger(\"mouseenter\")",
+    unhoverCloseBtn = "jQuery(\".flc-quickSetStrip > div:last-of-type\").trigger(\"mouseleave\")",
+    focusCloseBtn = "var event = jQuery.Event(\"keyup\"); event.shiftKey = true; event.key = \"Tab\"; jQuery(\".flc-quickSetStrip > div:first-of-type\").trigger(event)",
+
+    clickCloseBtn = "jQuery(\".flc-quickSetStrip > div:last-of-type\").click()",
+    hoverLanguageBtn = "jQuery(\".flc-quickSetStrip > div:first-of-type\").trigger('mouseenter')",
+    clickLanguageBtn = "jQuery(\".flc-quickSetStrip > div:first-of-type\").click()",
+    clickScreenZoomBtn = "jQuery(\".flc-quickSetStrip > div:nth-of-type(2)\").click()",
+    clickAppTextZoomBtn = "jQuery(\".flc-quickSetStrip > div:nth-of-type(3)\").click()",
+    clickReadAloudBtn = "jQuery(\".flc-quickSetStrip > div:nth-of-type(5)\").click()",
+    clickMoreBtn = "jQuery(\".flc-quickSetStrip > div:nth-last-of-type(6)\").click()",
+    clickSaveBtn = "jQuery(\".flc-quickSetStrip > div:nth-last-of-type(5)\").click()",
+    clickUndoBtn = "jQuery(\".flc-quickSetStrip > div:nth-last-of-type(4)\").click()",
+    clickPspBtn = "jQuery(\".flc-quickSetStrip > div:nth-last-of-type(3)\").click()",
+    clickResetAllBtn = "jQuery(\".flc-quickSetStrip > div:nth-last-of-type(2)\").click()",
     getQssSettingsList = "(function getItems() { var repeater = fluid.queryIoCSelector(fluid.rootComponent, 'gpii.psp.repeater')[0]; return repeater.model.items; }())";
 
 // QSS Widgets related
 var checkIfMenuWidget = "jQuery('.flc-qssMenuWidget').is(':visible');",
     checkIfStepperWidget = "jQuery('.flc-qssStepperWidget').is(':visible');",
-    clickMenuWidgetItem = "jQuery('.flc-qssWidgetMenu-item:nth-child(2)').click()",
+    clickMenuWidgetItem = "jQuery('.flc-qssWidgetMenu-item:nth-of-type(2)').click()",
     clickIncreaseBtn = "jQuery('.flc-qssStepperWidget-incBtn').click()",
-    clickDecreaseBtn = "jQuery('.flc-qssStepperWidget-decBtn').click()";
+    clickDecreaseBtn = "jQuery('.flc-qssStepperWidget-decBtn').click()",
+    clickToggleBtn = "jQuery('.flc-switchUI-control').click()";
 
 // Generic
 var closeClosableDialog = "jQuery(\".flc-closeBtn\").click()";
 
+var openReadAloudMenuSeqEl = {
+    task: "gpii.test.executeJavaScript",
```

Comment:
This function has an unhelpfully generic name. Perhaps it would more accurately be something like gpii.test.executeWebContentsJavaScript? Even this doesn't seem quite right since it doesn't execute arbitrary code. Maybe gpii.test.invokeFunctionInWebContents?

Reply:
Yes, it really sounds too generic. We were following the naming of the Electron WebContents api.
Well, currently this method simply executes a js as a string so I think the first name fits better

On the other hand it seems more convenient to use functions instead of a script strings. Lately, we've been using a "functions approach", e.g. [here](https://github.com/GPII/gpii-app/pull/77/files#diff-320839322d3d294a4886a140356266feR139), so I can go ahead and add such a utility function

Reply:
Yeah, that sounds good. Something with the signature of and similar name to https://docs.fluidproject.org/infusion/development/CoreAPI.html#fluidinvokeglobalfunctionfunctionpath-args , only acting in the webContents, would be a useful API

Reply:
I also put some thought in a way to **provide arguments to the invoked function** but it's not that straightforward as the function code is sent to the renderer as a string.
 
We could, do something like:
```
function argumentsToString (args) {
    var varsSrting = "",
        varTemplate = "%name=%value;";
    fluid.each(args, function (value, name) {
        varsSrting += fluid.stringTemplate(varTemplate, {name: name, value: value});
    });
 
    return varsSrting;
};

gpii.test.toIIFEString = function (func, args) {
     return fluid.stringTemplate("(%func)(%args)"), { func: func.toString(), args: fluid.keys(args).join(",").toString();
}
 
gpii.test.invokeFunctionInWebContents = function (dialog, func, args) {
    var varsString = argumentsToString (args),
        command = fluid.stringTemplate("%arguments;%func", {
            func: gpii.test.toIIFEString(func, args), // adds arg keys to the function invoke string
            arguments: varsSrting
        });
    return dialog.webContents.executeJavaScriptInWebContents(command, true);
};
 
gpii.test.invokeFunctionInWebContents(dialog, someFuncName, { arg1: "value1", arg2: 2 })
```
This will add variable declarations at the beginning of the code string which are also added to the IIFE part of the string.
This should work and would add some flexibility but I have mixed feelings about it...

Reply:
I think this could be simplified by just embedding the argument values directly within the function invocation, especially since we don't want to dump a lot of extra conflicting definitions into what I imagine will be the global scope of the browser

Reply:
Yep, that sounds more reasonable. I've come up with this idea from a different angle and totally missed this possibility. So it would look something like:
```
gpii.test.toIIFEString = function (func, args) {
     return fluid.stringTemplate("(%func)(%args)"), { 
                  func: func.toString(),
                  args: args.join(",").toString(); // I think special treatment would be needed for string type - probably addition of quotes
} 
```